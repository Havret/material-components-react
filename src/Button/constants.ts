const BUTTON = 'mdc-button';
const BUTTON_ICON = 'mdc-button__icon';
const BUTTON_RAISED = 'mdc-button--raised';
const BUTTON_UNELEVATED = 'mdc-button--unelevated';
const BUTTON_STROKED = 'mdc-button--stroked';
const BUTTON_DENSE = 'mdc-button--dense';
const BUTTON_COMPACT = 'mdc-button--compact';

export { BUTTON, BUTTON_ICON, BUTTON_RAISED, BUTTON_UNELEVATED, BUTTON_STROKED, BUTTON_DENSE, BUTTON_COMPACT }