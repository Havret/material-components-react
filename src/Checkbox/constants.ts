export const ROOT = 'mdc-checkbox';
export const UPGRADED = 'mdc-checkbox--upgraded';
export const NATIVE_CONTROL = 'mdc-checkbox__native-control';
export const DISABLED = 'mdc-checkbox--disabled';
export const BACKGROUND = 'mdc-checkbox__background';
export const CHECKMARK = 'mdc-checkbox__checkmark';
export const CHECKMARK_PATH = 'mdc-checkbox__checkmark__path';
export const MIXEDMARK = 'mdc-checkbox__mixedmark';

export const ANIM_UNCHECKED_CHECKED = 'mdc-checkbox--anim-unchecked-checked';
export const ANIM_UNCHECKED_INDETERMINATE = 'mdc-checkbox--anim-unchecked-indeterminate';
export const ANIM_CHECKED_UNCHECKED = 'mdc-checkbox--anim-checked-unchecked';
export const ANIM_CHECKED_INDETERMINATE = 'mdc-checkbox--anim-checked-indeterminate';
export const ANIM_INDETERMINATE_CHECKED = 'mdc-checkbox--anim-indeterminate-checked';
export const ANIM_INDETERMINATE_UNCHECKED = 'mdc-checkbox--anim-indeterminate-unchecked';

export const ANIM_END_LATCH_MS = 100;

export enum CheckState {
    checked,
    unchecked,
    indeterminate
}