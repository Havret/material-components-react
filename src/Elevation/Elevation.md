Shadows provide important visual cues about objects’ depth and directional movement. They are the only visual cue indicating the amount of separation between surfaces. An object’s elevation determines the appearance of its shadow. The elevation values are mapped out in a "z-space" and range from 0 to 24.

```js
initialState = { elevation: 0 };
<div className="demo-surfaces">    
    <Elevation z={10}>10dp</Elevation>    
    <Elevation 
        z={state.elevation}
        transition={true}
        onMouseOver={() => setState({elevation: 24})}
        onMouseOut={() => setState({elevation: 0})}
        >
            Hover Me 
            {state.elevation}dp            
    </Elevation>
</div>
```