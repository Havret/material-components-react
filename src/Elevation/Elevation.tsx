import * as React from 'react';
import { HTMLProps, ReactNode } from 'react';

interface ElevationProps extends HTMLProps<HTMLInputElement> {
    /**
     * The content of the Elevation.
     */
    children?: ReactNode;
    /**
     * Addition CSS classes that will be added to the root element.
     */
    z: 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 | 13 | 14 | 15 | 16 | 17 | 18 | 19 | 20 | 21 | 22 | 23 | 24;
    /**
     * Allows for smooth transitions between elevations when the z value changes.
     */
    transition?: boolean;
}

class Elevation extends React.Component<ElevationProps, {}> {
    render() {
        const {className} = this.props;

        return (
            <div></div>
        );
    }
}

export { Elevation as default, Elevation };