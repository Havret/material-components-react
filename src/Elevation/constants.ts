export const ELEVATION = 'mdc-elevation--z';
export const ELEVATION_TRANSITION = 'mdc-elevation-transition';