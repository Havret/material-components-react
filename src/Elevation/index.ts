export { default, Elevation, ElevationProps } from './Elevation';