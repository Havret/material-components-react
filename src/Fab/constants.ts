export const FAB = 'mdc-fab';
export const FAB_ICON = 'mdc-fab__icon';
export const FAB_MINI = 'mdc-fab--mini';
export const FAB_EXITED = 'mdc-fab--exited';