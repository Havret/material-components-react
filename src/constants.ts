export const MATERIAL_ICONS = 'material-icons';