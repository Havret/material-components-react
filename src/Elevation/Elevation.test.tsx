import { configure, shallow } from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import * as React from 'react';
import toJson from 'enzyme-to-json';
import { Elevation, ElevationProps } from './Elevation';
import { ELEVATION, ELEVATION_TRANSITION } from './constants';

describe('Elevation', () => {
    configure({adapter: new Adapter()});

    const setUp = (props: ElevationProps, content: React.ReactNode = 'Elevation Content') => {
        return shallow(<Elevation {...props}>{content}</Elevation>);
    };

    test('It should have mdc classes', () => {
        const wrapper = setUp({z: 2});
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('It should set elevation', () => {
        const wrapper = setUp({z: 10});
        expect(wrapper.hasClass(`${ELEVATION}${10}`)).toBeTruthy();
    });

    test('It should not add elevation transition class', () => {
        const wrapper = setUp({z: 10, transition: false});
        expect(wrapper.hasClass(ELEVATION_TRANSITION)).toBeFalsy();
    });

    test('It should add elevation transition class', () => {
        const wrapper = setUp({z: 10, transition: true});
        expect(wrapper.hasClass(ELEVATION_TRANSITION)).toBeTruthy();
    });

    test('It should add extra class to the root element', () => {
        const extraClass = 'Extra Class';
        const wrapper = setUp({z: 10, className: extraClass});
        expect(wrapper.hasClass(extraClass)).toBeTruthy();
    });
});
