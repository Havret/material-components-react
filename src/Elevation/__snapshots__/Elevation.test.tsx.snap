// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Elevation It should have mdc classes 1`] = `
<div
  className="mdc-elevation--z2"
>
  Elevation Content
</div>
`;
